src/DDTrace/Span.php

```diff
@@ -400,4 +418,14 @@ public function isTraceAnalyticsCandidate()
     {
         return false;
     }
+
+    public function __destruct()
+    {
+        if (!$this->isFinished()) {
```

Comment:
Yes. isFinished() directly reads the duration of the internal span (via the magic __get in Data\Span). It is always accurate.

Reply:
Thanks for clarifying.

Did I ever mention that I hate magic methods 😄  (here is a [link](https://github.com/DataDog/dd-trace-php/blob/e863039f30e216aef1341646b76bab0735ade8a9/src/api/Data/Span.php#L43) to what Bob is referring to for future reference). Note that this is not a critique of this specific use case, since here we are building an adapter we are justified if we use some tricks.